lib/blocktree/database.go

```diff
@@ -0,0 +1,138 @@
+package blocktree
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"io"
+	"math/big"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+// Store stores the blocktree in the underlying db
+func (bt *BlockTree) Store() error {
+	if bt.db == nil {
+		return errors.New("blocktree db is nil")
+	}
+
+	enc, err := bt.Encode()
+	if err != nil {
+		return err
+	}
+
+	return bt.db.Put(common.BlockTreeKey, enc)
+}
+
+// Load loads the blocktree from the underlying db
+func (bt *BlockTree) Load() error {
+	if bt.db == nil {
+		return errors.New("blocktree db is nil")
+	}
+
+	enc, err := bt.db.Get(common.BlockTreeKey)
+	if err != nil {
+		return err
+	}
+
+	return bt.Decode(enc)
+}
+
+// Encode recursively encodes the block tree
+// enc(node) = [32B block hash + 8B arrival time + 8B num children n] | enc(children[0]) | ... | enc(children[n-1])
+func (bt *BlockTree) Encode() ([]byte, error) {
+	return encode(bt.head, []byte{})
+}
+
+func encode(n *node, enc []byte) ([]byte, error) {
```

Comment:
missing method comment, would be nice to clearly differentiate between `Encode` and `encode`

Reply:
will add